### PR TITLE
Implement corner collision resolution

### PR DESCRIPTION
**Pull request by Tyler-IN**
change CalculateNormal to apply rounded collision resolution to corners

reduce concurrency on concurrent dictionaries

add TODO for container mass

minor refactor of GetCollidingEntities, and don't re-enumerate modifiers in an inner loop

### CODE REVIEW

File: Robust.Shared/Physics/PhysicsManager.cs [hunk 1]
```diff
@@ -52,22 +55,29 @@ public bool IsWeightless(GridCoordinates gridPosition)
...
+            var rightDist = targetBox.Right - sourceBox.Left;
+            var x = leftDist - rightDist;
+            // Y is the axis of separation
+            var bottomDist = sourceBox.Top - targetBox.Bottom;
+            var topDist = targetBox.Top - sourceBox.Bottom;
+            var y = bottomDist - topDist;
+
+            // octagonal surface normals; create 8 regions, a third of each of the 4 faces is flat
+            // corners are treated as curved during resolution, making players feel somewhat pill shaped
+            const float faceToCornerBias = 1/3f; // higher is more square, higher is more pill shaped
```
Comment (RemieRichards) [inline Robust.Shared/Physics/PhysicsManager.cs]: comment seems wrong

File: Robust.Client/Console/Commands/Debug.cs [hunk 2]
```diff
@@ -758,7 +758,7 @@ internal class ReloadShadersCommand : IConsoleCommand
 
         public static Dictionary<string, FileSystemWatcher> _watchers;
 
-        public static ConcurrentDictionary<string, bool> _reloadShadersQueued = new ConcurrentDictionary<string, bool>();
+        public static ConcurrentDictionary<string, bool> _reloadShadersQueued = new ConcurrentDictionary<string, bool>(1,4);
```
Comment (ComicIronic) [inline Robust.Client/Console/Commands/Debug.cs:761]: Is there any particular reason for choosing 4 as the initial capacity?

File: Robust.Shared/Physics/PhysicsManager.cs (hunk 1 above)
Comment (Tyler-IN) [reply Robust.Shared/Physics/PhysicsManager.cs]: how so?

File: Robust.Client/Console/Commands/Debug.cs (hunk 2 above)
Comment (Tyler-IN) [reply Robust.Client/Console/Commands/Debug.cs:761]: It's a small power of 2

File: Robust.Shared/Physics/PhysicsManager.cs (hunk 1 above)
Comment (Tyler-IN) [reply Robust.Shared/Physics/PhysicsManager.cs]: oh - higher is more square, lower is more pill shaped

Comment (Tyler-IN) [reply Robust.Shared/Physics/PhysicsManager.cs]:
```suggestion
            const float faceToCornerBias = 1/3f; // higher is more square, lower is more pill shaped
```

File: Robust.Client/Console/Commands/Debug.cs (hunk 2 above)
Comment (Tyler-IN) [reply Robust.Client/Console/Commands/Debug.cs:761]: (Smaller than the default size, still scales acceptably)

**Review by PJB3005 — COMMENTED** [review]
It feels like a terrible idea to make this enabled for every single bounding box unconditionally.

**Pull request closed without merging**